tools/wltests/test_series

```diff
@@ -786,12 +786,25 @@ match_sha1() {
 	COMMIT_SHA1=$1
 
 	c_info "Current kernel: "
-	CURRENT=$($ADB shell 'uname -a')
-	c_info "   $CURRENT"
+	c_info "   $($ADB shell 'uname -a')"
 
-	[[ $CURRENT = *$COMMIT_SHA1* ]] || return $EAGAIN
+	CURRENT=$($ADB shell 'uname -r | awk -F "-" "{print \$NF}"')
+	CURRENT=${CURRENT:1}
 
-	return $OK
+	[[ $COMMIT_SHA1 =~ $CURRENT* ]] && return $OK
+
+	# It may happen that a tag pointing to a given commit has
+	# a different SHA1 (signed tag), so check if this this is
+        # the case.
+	TAG_NAME=$(git -C $KERNEL_SRC describe --exact-match $COMMIT_SHA1 2> /dev/null) || return $EAGAIN
+	TAG_SHA1=$(git -C $KERNEL_SRC show-ref --hash $TAG_NAME)
+
+	if [[ $TAG_SHA1 =~ $CURRENT* ]]; then
```

Comment:
Not entirely sure, but could it happen that we find a TAG which still do not match the CURRENT?
In that case, is not more generic to always report the tag found?

```python
TAG_SHA1=$(git -C $KERNEL_SRC show-ref --hash $TAG_NAME)
c_info "   Corresponding tag: $TAG_NAME"
[[ $TAG_SHA1 =~ $CURRENT* ]] && return $OK

return $EAGAIN
```

As per above comment, I would put this last chunk into a dedicated patch, since this is really adding a new functionality while the previous chunk is improving an existing one.... as a bonus you also get two more patches in LISA :-P

Reply:
It does not change too much, but OK, your solution would be more verbose.
BUT since the last log message shows the kernel version of the device, this would be misleading.

I'll try to change the message text accordingly.